packages/manager/apps/web/client/app/private-database/whitelist/private-database-whitelist.controller.js

```diff
@@ -192,5 +205,53 @@ angular.module('App').controller(
         );
       }
     }
+
+    getWebhostingNetworkStatus() {
+      this.whitelistService
+        .getWebhostingNetworkStatus(this.serviceName)
+        .then((res) => {
+          this.$scope.webhostingNetworkStatus = res;
+
+          this.$scope.isWebhostingNetworkDisplayOn = [
+            WEBHOSTING_NETWORK_STATUS.enabling,
+            WEBHOSTING_NETWORK_STATUS.enabled,
+          ].includes(this.$scope.webhostingNetworkStatus);
+
+          this.$scope.isWebhostingNewtworkChangeDisabled = [
+            WEBHOSTING_NETWORK_STATUS.enabling,
+            WEBHOSTING_NETWORK_STATUS.disabling,
+          ].includes(this.$scope.webhostingNetworkStatus);
+        })
+        .catch((err) => this.alerter.error(err));
+    }
+
+    onWebhostingNetworkStatusChange() {
+      switch (this.$scope.webhostingNetworkStatus) {
+        case WEBHOSTING_NETWORK_STATUS.disabled:
+          this.whitelistService.enableWebhostingNetwork(this.serviceName);
+          break;
+        case WEBHOSTING_NETWORK_STATUS.enabled:
+          this.whitelistService.disableWebhostingNetwork(this.serviceName);
+          break;
+        default:
+          return;
+      }
+
+      this.$scope.isWebhostingNewtworkChangeDisabled = true;
+      this.alerter.success(
+        this.$translate.instant(
+          'private_database_tabs_whitelist_webhosting_network_access_success',
+        ),
+        this.$scope.alerts.main,
+      );
+    }
+
+    handleWebhostingNetworkUpdate(state, message) {
+      this.getWebhostingNetworkStatus();
```

Comment:
Not an issue that this is asynchronous ?

Reply:
Seen together, all good.

Reply:
Seen together, all good.